flocx_market/tests/unit/conftest.py

```diff
@@ -33,14 +36,15 @@ def client(app):
 
 @pytest.fixture()
 def db(app):
-    orm.create_all()
-    yield orm
-    orm.drop_all()
+    db_api.setup_db()
+    yield db_api
+    db_api.get_session().query(db_models.Offer).delete()
```

Comment:
This fixture is used by all the database tests...not just offers. Do we really need to explicitly delete objects like this? Can we just call `drop_db()` and be done with it?

Reply:
You'd think so, but apparently the data remains... somehow... and causes integrity errors if you try and insert the same data into the database in subsequent tests. I checked with a few other OpenStack projects to see how they dealt with the issue, and discovered that most OpenStack projects don't test the database API. The one that I found that does has this:

https://github.com/openstack/mistral/blob/master/mistral/tests/unit/base.py#L280

Reply:
In that case, shouldn't this also delete Bids in the same way?

Reply:
Whoops, you're absolutely right; added this in. I think the tests still passed because Bids have no unique constraints.

Reply:
It bothers me seeing the models called out explicitly like this (because of how we potentially need to update this fixture every time we add a new model). How about this instead?

```diff
diff --git a/flocx_market/tests/unit/conftest.py b/flocx_market/tests/unit/conftest.py
index b88d6b02..22ec96c0 100644
--- a/flocx_market/tests/unit/conftest.py
+++ b/flocx_market/tests/unit/conftest.py
@@ -38,8 +38,6 @@ def client(app):
 def db(app):
     db_api.setup_db()
     yield db_api
-    db_api.get_session().query(db_models.Offer).delete()
-    db_api.get_session().query(db_models.Bid).delete()
     db_api.drop_db()
 
 
@@ -47,5 +45,7 @@ def db(app):
 def session(db):
     session = db.get_session()
     yield session
-    # Guess what doesn't work with oslo db sessions...
-    # session.rollback()
+    session.close()
+    engine = db_api.get_facade().get_engine()
+    for table in db_models.Base.metadata.sorted_tables:
+        engine.execute(table.delete())
```

Reply:
You're right, that works great! Thanks for the code snippet - I've updated the PR.